client/connections/ConnectionToCore.ts

```diff
@@ -169,8 +170,17 @@ export default abstract class ConnectionToCore extends TypedEventEmitter<{
   }
   ///////  SESSION FUNCTIONS  //////////////////////////////////////////////////////////////////////////////////////////
 
-  public async createSession(options: ISessionCreateOptions): Promise<CoreSession> {
+  public async createSession(
+    options: Omit<ISessionCreateOptions, 'externalIds' | 'sessionId'> &
+      Pick<IHeroCreateOptions, 'externalIds' | 'sessionId'>,
+  ): Promise<CoreSession> {
     try {
+      if (options.sessionId) options.sessionId = await options.sessionId;
+      if (options.externalIds) {
+        for (const [key, value] of Object.entries(options.externalIds)) {
+          options.externalIds[key] = await value;
```

Comment:
Do we really want external Ids to be Promisable? Do we have an immediate need?

Reply:
preferably not... but our only use case is a databox promise for sessionId

Reply:
That's because the sessionId is created in Databox Core. Should we just create it on client and pass it through?

Reply:
I haven't come around to the idea of client created session ids quite yet. Do you think that's what you'd prefer we do?

Reply:
I don't see any harm. The only risk is someone creates a sessionId that conflicts, but I assume the SessionsDb will throw a unique error.

Reply:
We do use that sessionId in various places in the codebase as names for file paths, so it also has to get checked to make sure it can actually be a filename.

Do you think this kind of change would go in this PR? Or as a follow-on?

Reply:
Your call.